qemu/tests/win_driver_install_by_installer.py

```diff
@@ -0,0 +1,329 @@
+
+        uninst_cmd = "%s remove %s" % (devcon_path, device_hwid)
+        status, output = session.cmd_status_output(uninst_cmd, inst_timeout)
+        # acceptable status: OK(0), REBOOT(1)
+        if status > 1:
+            test.error("Failed to uninstall driver '%s', details:\n"
+                       "%s" % (driver_name, output))
+
+    inst_timeout = int(params.get("driver_install_timeout", INSTALL_TIMEOUT))
+    chk_timeout = int(params.get("chk_timeout", 240))
+    installer_pkg_check_cmd = params["installer_pkg_check_cmd"]
+
+    # qemu version check test config
+    qemu_ga_pkg = params["qemu_ga_pkg"]
+    gagent_pkg_info_cmd = params["gagent_pkg_info_cmd"]
+    gagent_uninstall_cmd = params["gagent_uninstall_cmd"]
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login()
+
+    devcon_path = utils_misc.set_winutils_letter(session,
+                                                 params["devcon_path"])
+    status, output = session.cmd_status_output("dir %s" % devcon_path,
+                                               timeout=OPERATION_TIMEOUT)
+    if status:
+        test.error("Not found devcon.exe, details: %s" % output)
+
+    if params.get("run_install_test", "no") == "yes":
+
+        driver_name_list = ['netkvm', 'viorng', 'vioser',
+                            'balloon', 'pvpanic', 'vioinput',
+                            'viofs', 'viostor', 'vioscsi', 'qemufwcfg']
+
+        device_hwid_list = ['"PCI\\VEN_1AF4&DEV_1000" "PCI\\VEN_1AF4&DEV_1041"',
+                            '"PCI\\VEN_1AF4&DEV_1005" "PCI\\VEN_1AF4&DEV_1044"',
+                            '"PCI\\VEN_1AF4&DEV_1003" "PCI\\VEN_1AF4&DEV_1043"',
+                            '"PCI\\VEN_1AF4&DEV_1002" "PCI\\VEN_1AF4&DEV_1045"',
+                            '"ACPI\\QEMU0001"', '"PCI\\VEN_1AF4&DEV_1052"',
+                            '"PCI\\VEN_1AF4&DEV_105A"',
+                            '"PCI\\VEN_1AF4&DEV_1001" "PCI\\VEN_1AF4&DEV_1042"',
+                            '"PCI\\VEN_1AF4&DEV_1004" "PCI\\VEN_1AF4&DEV_1048"',
+                            '"ACPI\\QEMU0002"']
+
+        device_name_list = ["Red Hat VirtIO Ethernet Adapter", "VirtIO RNG Device",
+                            "VirtIO Serial Driver", "VirtIO Balloon Driver",
+                            "QEMU PVPanic Device", "VirtIO Input Driver",
+                            "VirtIO FS Device", "Red Hat VirtIO SCSI controller",
+                            "Red Hat VirtIO SCSI pass-through controller",
+                            "QEMU FWCfg Device"]
+        for driver_name, device_name, device_hwid in zip(driver_name_list,
+                                                         device_name_list, device_hwid_list):
+            uninstall_driver(session, driver_name, device_name, device_hwid)
+
+        error_context.context("Install 'qemu-guest-agent' package in guest.",
+                              logging.info)
```

Comment:
I suggest move qga package step ahead of uninstall drivers step.
Since qga feature own is considering adding serial driver install if it's not installed.
That means your serial uninstall step will be meaningless.

Even though he will not modify your part, I still suggest to modify it since the logic of qga usage requirement is serial driver installed.